AVADA.pm

```diff
@@ -0,0 +1,234 @@
+  $self->expand_right(0);
+
+  $self->get_user_params();
+  my $param_hash = $self->params_to_hash();
+  my $file = $param_hash->{file};
+  die "\n  ERROR: No file specified\nTry using 'AVADA,file=path/to/file.tsv.gz'\n" unless defined($file);
+  $self->add_file($file);
+  $self->{feature_match_by} = $param_hash->{feature_match_by}; 
+  $self->{original_variant_string} = $param_hash->{original_variant_string}; 
+
+  if ($self->{original_variant_string}) {
+    $self->{config}->{refseq} = 1 unless $self->{config}->{database} == 1 ; # when using db, refseq is not enabled in spite of forcing in code
+    die "\n ERROR: Matching by Refseq ID requires the option --refseq when using the database  \n" if $self->{config}->{refseq} == 0 && $self->{config}->{database} == 1;
+  }
+  return $self;
+}
+
+sub run {
+  my ($self, $tva) = @_;
+
+  return {} if grep {$_->SO_term eq 'downstream_gene_variant' || $_->SO_term eq 'upstream_gene_variant'} @{$tva->get_all_OverlapConsequences};
+
+  my $vf = $tva->variation_feature;
+  my $tv = $tva->transcript_variation;
+
+  my ($vf_start, $vf_end) = ($vf->{start}, $vf->{end});
+  ($vf_start, $vf_end) = ($vf_end, $vf_start) if ($vf_start > $vf_end);
+
+  my $transcript = $tva->transcript;
+  my $alt_allele = $tva->variation_feature_seq;
+  my %output;
+  my $refseq_transcript;
+  my $refseq_protein;
+  my $gene_key = "AVADA_GENE_SYMBOL";
+  my @data;
+
+  if (!defined($self->{feature_match_by})){
+    @data = @{$self->get_data($vf->{chr}, $vf_start, $vf_end)}; 
+  }
+  elsif ( $self->{feature_match_by} eq "gene_symbol"){
+    @data = grep {
+    $_->{AVADA_GENE_SYMBOL} eq $transcript->{_gene_symbol}
+    }@{$self->get_data($vf->{chr}, $vf_start, $vf_end)};
+  }
+  elsif ( $self->{feature_match_by} eq "ensembl_gene_id" ){
+  @data = grep {
+    $_->{AVADA_ENSEMBL_ID} eq $transcript->{_gene_stable_id}
+    }@{$self->get_data($vf->{chr}, $vf_start, $vf_end)};
+  $gene_key = "AVADA_ENSEMBL_ID";
+  }
+  else{
+    die("ERROR: feature_match_by can only take one of the options gene_symbol|ensembl_gene_id ");
+  }
+  if ( $self->{original_variant_string} ){
+    $refseq_transcript = $transcript->{stable_id};
+    $refseq_protein = $transcript->translation->{stable_id} if defined($transcript->translation);
+    $refseq_protein =~ s/cds-// if $self->{config}->{database} == 1;
+    @data = grep {
+      split(".",$_->{AVADA_REFSEQ_ID}) eq split(".",$refseq_transcript) || 
+      $_->{AVADA_REFSEQ_ID} eq $refseq_protein 
+      } @data;
+  }
+
+  return {} unless scalar @data;
+
+  my $pmid_string;
+  my $output_key;
+  my %seen;
+  foreach my $data_value (uniq @data) {
+    next unless $alt_allele eq $data_value->{AVADA_ALT};
+    $output{"AVADA_GENE_ID"} =  $data_value->{$gene_key};
+    if ($self->{original_variant_string})
+    {
+    # Output is in the format "PMID%RefSeq_ID%Variant_string"
+    my $pmid_variant = $data_value->{AVADA_PMID}."%".$data_value->{AVADA_REFSEQ_ID}."%".$data_value->{AVADA_VARIANT_STRING};
```

Comment:
This could be improved to return an array for JSON:
```suggestion
    my $pmid_variant = [$data_value->{AVADA_PMID}, $data_value->{AVADA_REFSEQ_ID}, $data_value->{AVADA_VARIANT_STRING}];
    $pmid_variant = join('%', @$pmid_variant) if not $self->{config}->{output_format} eq 'json';
```

Reply:
Thanks @nuno-agostinho , modified the output for json to send a hash, please let me know if you have suggestions